ctapipe_io_magic/__init__.py

```diff
@@ -596,66 +603,142 @@ def parse_data_info(self):
+                        )
+                        has_prescaler_info = False
 
-                if (
-                    prescaler == prescaler_mono_nosumt
-                    or prescaler == prescaler_mono_sumt
-                ):
-                    stereo = False
-                    hast = False
-                elif prescaler == prescaler_stereo:
-                    stereo = True
-                    hast = False
-                elif prescaler == prescaler_hast:
-                    stereo = True
-                    hast = True
-                else:
-                    stereo = True
-                    hast = False
+                if not has_trigger_info or not has_prescaler_info:
+                    if stereo_prev is not None and hast_prev is not None:
+                        logger.warning(
+                            "Assuming previous subrun information for trigger settings."
+                        )
+                        stereo = stereo_prev
+                        hast = hast_prev
+                    else:
+                        logger.warning("Assuming standard stereo data.")
+                        stereo = True
+                        hast = False
+
+                if has_prescaler_info:
+                    prescaler = None
+                    prescaler_size = prescaler_array.size
+                    if prescaler_size > 1:
+                        prescaler = list(prescaler_array[1])
+                    elif prescaler_size == 1:
+                        prescaler = list(prescaler_array[0])
+                    else:
+                        logger.warning(
+                            f"No prescaler info found in {rootf.file_path}."
+                        )
+                        if stereo_prev is not None and hast_prev is not None:
+                            logger.warning(
+                                "Assuming previous subrun information for trigger settings."
+                            )
+                            stereo = stereo_prev
+                            hast = hast_prev
+                        else:
+                            logger.warning("Assuming standard stereo data.")
+                            stereo = True
+                            hast = False
+
+                    if prescaler is not None:
+                        if (
+                            prescaler == prescaler_mono_nosumt
+                            or prescaler == prescaler_mono_sumt
+                        ):
+                            stereo = False
+                            hast = False
+                        elif prescaler == prescaler_stereo:
+                            stereo = True
+                            hast = False
+                        elif prescaler == prescaler_hast:
+                            stereo = True
+                            hast = True
+                        else:
+                            stereo = True
```

Comment:
this means some unusual prescaler setting that in principle should not happen in the data. Maybe worth to put a warning information listing what kind of prescaler was read in.

Reply:
Done.